pkg/quarterdeck/db/models/users.go

```diff
@@ -187,6 +188,48 @@ func GetUserByToken(ctx context.Context, token string) (u *User, err error) {
 	return u, nil
 }
 
+// GetUser by delete confirmation token by doing a join with the organization_users
+// table. This returns an error if the token is invalid or expired.
+func GetUserByDeleteToken(ctx context.Context, userID, orgID any, token string) (u *User, err error) {
+	u = &User{}
+	if u.ID, err = ulids.Parse(orgID); err != nil {
+		return nil, err
+	}
+
+	orgUser := &OrganizationUser{}
+	if orgUser.OrgID, err = ulids.Parse(orgID); err != nil {
+		return nil, err
+	}
+
+	var tx *sql.Tx
+	if tx, err = db.BeginTx(ctx, &sql.TxOptions{ReadOnly: true}); err != nil {
+		return nil, err
+	}
+
+	if err = tx.QueryRow(getUserDeleteTokenSQL, sql.Named("userID", u.ID), sql.Named("orgID", orgUser.OrgID), sql.Named("token", token)).Scan(&u.ID, &token); err != nil {
```

Comment:
This line is replacing the token passed in by the user - but presumably they are the same input? Is there a reason the token is being fetched from the database again?

Reply:
Right, since the lookup already includes the token I'll remove it from the SELECT to reduce the confusion.